.github/workflows/job-precommit-java-examples-dataflow.yml

```diff
@@ -0,0 +1,90 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# To learn more about GitHub Actions in Apache Beam check the CI.md
+# Runs PreCommit Java Examples Dataflow scheduled on cron, on push to master and on pull request
+
+name: PreCommit Java Examples Dataflow
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['model/**',
+            'sdks/java/**',
+            'runners/google-cloud-dataflow-java/**',
+            'examples/java/**',
+            'examples/kotlin/**',
+            'release/**']
+permissions: read-all
+
+
+jobs:
+  set-properties:
+    runs-on: [self-hosted, ubuntu-20.04]
+    outputs:
+      properties: ${{ steps.test-properties.outputs.properties }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
+          submodules: false
+      - id: test-properties
+        uses: ./.github/actions/setup-default-test-properties
+
+  precommit-java-examples-dataflow:
+    needs: set-properties
+    name: Run PreCommit Java Examples Dataflow
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix: #JDK11 and 17 compilation will fail due to changes in ByteBuffer class since JDK9
+              #TODO: Retest once issue https://github.com/apache/beam/issues/23068 gets resolved.
```

Comment:
Did this problem exist in the Jenkins version? If not, why are we seeing it now?

Reply:
In jenkins, the jar files are being compiled with JDK8 and executed with JDK11 or 17.
These errors crop up when compiling with JDK11/17.

JDK 9 introduced restrictions on the method visibility for the ByteBuffer class, resulting in NoSuchMethodError exceptions at compile time.
We tried going the same route in GHA, that is, adding compile-time flags in Gradle (--add-exports and --add-opens) to circumvent the ByteBuffer methods visibility restrictions put in place in JDK9, with no success as of yet.

I opened this issue https://github.com/apache/beam/issues/23068 to keep track of this.

JDK16 did something similar but that change affects the ErrorProne library. We also tried applying the same --add-exports and --add-opens flags, but they were not being picked up.

Issue https://github.com/apache/beam/issues/23066 as opened to track that one.

We consulted this with Kerry, and at that moment we decided to keep the associated jenkins jobs and exclude that matrix from the GHA workflows.

Reply:
Ok, sounds good. As long as we're not replacing the jenkins jobs